kratos/utilities/discont_utils.h

```diff
@@ -1105,17 +1071,17 @@ class DiscontinuousShapeFunctionsUtilities
             array_1d<double, 3 > & center_position,
             const int i0, const int i1, const int i2, const int i3)
     {
-        double x10 = aux_coordinates(i1, 0) - aux_coordinates(i0, 0); //geom[1].X() - geom[0].X();
-        double y10 = aux_coordinates(i1, 1) - aux_coordinates(i0, 1); // geom[1].Y() - geom[0].Y();
-        double z10 = aux_coordinates(i1, 2) - aux_coordinates(i0, 2); // geom[1].Z() - geom[0].Z();
+        double x10 = aux_coordinates(i1, 0) - aux_coordinates(i0, 0);
+        double y10 = aux_coordinates(i1, 1) - aux_coordinates(i0, 1);
+        double z10 = aux_coordinates(i1, 2) - aux_coordinates(i0, 2);
 
-        double x20 = aux_coordinates(i2, 0) - aux_coordinates(i0, 0); // geom[2].X() - geom[0].X();
-        double y20 = aux_coordinates(i2, 1) - aux_coordinates(i0, 1); // geom[2].Y() - geom[0].Y();
-        double z20 = aux_coordinates(i2, 2) - aux_coordinates(i0, 2); // geom[2].Z() - geom[0].Z();
+        double x20 = aux_coordinates(i2, 0) - aux_coordinates(i0, 0);
+        double y20 = aux_coordinates(i2, 1) - aux_coordinates(i0, 1);
+        double z20 = aux_coordinates(i2, 2) - aux_coordinates(i0, 2);
 
-        double x30 = aux_coordinates(i3, 0) - aux_coordinates(i0, 0); // geom[3].X() - geom[0].X();
-        double y30 = aux_coordinates(i3, 1) - aux_coordinates(i0, 1); // geom[3].Y() - geom[0].Y();
-        double z30 = aux_coordinates(i3, 2) - aux_coordinates(i0, 2); // geom[3].Z() - geom[0].Z();
+        double x30 = aux_coordinates(i3, 0) - aux_coordinates(i0, 0);
+        double y30 = aux_coordinates(i3, 1) - aux_coordinates(i0, 1);
+        double z30 = aux_coordinates(i3, 2) - aux_coordinates(i0, 2);
 
         double detJ = x10 * y20 * z30 - x10 * y30 * z20 + y10 * z20 * x30 - y10 * x20 * z30 + z10 * x20 * y30 - z10 * y20 * x30;
         double vol = detJ * 0.1666666666666666666667;
```

Comment:
1.0/6.0 will do the same effect as compiler will replace it in compile time.

Reply:
Changed by / 6.0. Thanks @pooyan-dadvand!